packages/client/hooks/useGetRefHeight.ts

```diff
@@ -0,0 +1,23 @@
+import {RefObject, useLayoutEffect, useState} from 'react'
+import useResizeObserver from './useResizeObserver'
+
+const useGetRefHeight = <T extends number | string>(
+  ref: RefObject<HTMLDivElement>,
+  defaultHeight?: T,
+  observerRef?: RefObject<HTMLDivElement>
+) => {
+  const [height, setHeight] = useState(defaultHeight || 0)
+
+  const getHeight = () => {
+    const {current: el} = ref
+    const height = el?.clientHeight
+    if (!height) return
+    setHeight(height)
+  }
+
+  useLayoutEffect(getHeight, [ref])
+  useResizeObserver(getHeight, observerRef)
```

Comment:
We want to know the height of the group columns but we don't know what it is until it's opened. `useLayoutEffect` makes an initial calculation and `useResizeObserver` then corrects it if it's slightly off